core/src/main/java/tc/oc/pgm/map/MapInfoImpl.java

```diff
@@ -384,9 +393,23 @@ public VariantData(Element root, @Nullable Element variantEl) throws InvalidXMLE
         boolean override = XMLUtils.parseBoolean(Node.fromAttr(variantEl, "override"), false);
         this.mapName = (override ? "" : name + ": ") + variantEl.getTextNormalize();
         this.world = variantEl.getAttributeValue("world");
-        if (slug != null) slug += "_" + variantId;
+
+        String variantSlug = variantEl.getAttributeValue("slug");
+        if (variantSlug != null) slug = variantSlug;
+        else if (slug != null) slug += "_" + variantId;
+
+        Node minVerVariant = Node.fromAttr(variantEl, "min-server-version");
+        if (minVerVariant != null) minVer = minVerVariant;
+
+        Node maxVerVariant = Node.fromAttr(variantEl, "max-server-version");
+        if (maxVerVariant != null) minVer = minVerVariant;
```

Comment:
🚨💥 variant-level max-server-version setting is being straight up ignored
```suggestion
        if (maxVerVariant != null) maxVer = maxVerVariant;
```

testing diff over OCC PublicMaps:

```diff
diff --git a/ctf/directrix/map.xml b/ctf/directrix/map.xml
index 9811fc9e..525771c0 100644
--- a/ctf/directrix/map.xml
+++ b/ctf/directrix/map.xml
@@ -3,8 +3,8 @@
 <variant id="halloween" world="halloween" override="true">DirecTricks</variant>
 <variant id="christmas" world="christmas" override="true">DirectriXmas</variant>
 <variant id="classic" world="classic">Classic</variant>
-<variant id="king_ranch" world="king_ranch" override="true">King Ranch</variant>
-<variant id="king_ranch_og" world="king_ranch_og" override="true">King Ranch: OG</variant>
+<variant id="king_ranch" world="king_ranch" override="true" min-server-version="1.0" max-server-version="1.8.8">King Ranch</variant>
+<variant id="king_ranch_og" world="king_ranch_og" override="true" min-server-version="1.21">King Ranch: OG</variant>
 <unless variant="king_ranch,king_ranch_og">
     <version>1.1.5</version>
 </unless>
```